examples/nextjs/src/components/layout/navigation.js

```diff
@@ -21,7 +26,7 @@ function Navigation({ items, className }) {
                         <Link
                             href={{
                                 pathname: item.href,
-                                query: Object.fromEntries(searchParams.entries()) // We need to maintain the query params on the navigation, this way next loads the page with the same query params
+                                query: {...getVariantAsQueryParam(item.href, currentQueryParams)} // We need to maintain the query params on the navigation, this way next loads the page with the same query params
```

Comment:
Not sure if we should do this, we should make it as easy to implement as possible

Reply:
As we talk related to this, this is for not do a redirect. And all related to an assigned variant to the user only occurs in the Browser. That variant not come with `client.page.get` or `client.nav.get`